wave_lang/kernel/wave/codegen/handlers.py

```diff
@@ -1684,7 +1684,21 @@ def handle_select(emitter: WaveEmitter, node: fx.Node):
         raise ValidationError("Malformed arguments") from e
 
     unwrap = lambda x: cast_py_value(emitter, x).ir_value
-    selected = arith_d.select(unwrap(cond), unwrap(if_true), unwrap(if_false))
+    cond = unwrap(cond)
+    if_true = unwrap(if_true)
+    if_false = unwrap(if_false)
+
+    # Handle special scalar/rank-0 cases where if_true/if_false may be
+    # Dtype, vector<Dtype>, or vector<1xDtype>.
+    true_false_ranks = [get_rank(arg.type) for arg in (if_true, if_false)]
```

Comment:
This code is duplicated in several places in the handlers (see also in handle_generic_binary). Could you pull this into a common function that can be reused?